packages/core/src/machine.schema.json

```diff
@@ -0,0 +1,120 @@
+{
+  "type": "object",
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "definitions": {
+    "jsonFunction": {
+      "type": "object",
+      "properties": {
+        "$function": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "actionObject": {
+      "type": "object",
+      "properties": {
+        "type": { "type": "string" },
+        "exec": {
+          "oneOf": [
+            { "type": "string" },
+            { "$ref": "#/definitions/jsonFunction" }
+          ]
```

Comment:
Thanks @davidkpiano, (not only, but especially) this really really helps when using xstate definition as a compilation target! 😃